folsom/src/main/java/com/spotify/folsom/client/ascii/AsciiMemcacheDecoder.java

```diff
@@ -216,7 +216,17 @@ protected void decode(final ChannelHandlerContext ctx, final ByteBuf buf, final
           out.add(AsciiResponse.NOT_STORED);
           return;
         } else {
-          throw fail();
+          final String lineStr = toString(line);
+          switch (lineStr) {
+            case "SERVER_ERROR object too large for cache":
```

Comment:
Do we have any idea how stable these error messages are? If they change, we will change (back) the behaviour to closing the connection, which might be more unexpected than retaining it.

Reply:
Not the error message itself, but the `SERVER_ERROR` is in the protocol

https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L162-L179

Do you think it would be dangerous to catch all the `SERVER_ERROR`s and handle them the same way,  i.e. as succeeded futures with a `MemcacheStatus.SERVER_ERROR`? The protocol mentions that memcached itself closes the connection in case it's necessary, it might be good enough for us?

>In cases
  of severe server errors, which make it impossible to continue
  serving the client (this shouldn't normally happen), the server will
  close the connection after sending the error line.

Reply:
That could perhaps be done as a followup PR, I think we should try avoid to do too many things in the same change.